verifiers/internal/gha/provenance.go

```diff
@@ -396,6 +396,10 @@ func VerifyTag(prov iface.Provenance, expectedTag string) error {
 	}
 
 	tag, err := utils.TagFromGitRef(ref)
+	if tag == "" {
```

Comment:
`TagFromGitRef` validates that the tag isn't empty and starts with `refs/tag` so I don't think this is necessary. If the problem is with the error message itself maybe you can add this check to `utils.ValidateGitRef`?

Reply:
Ah, I see. This maybe when the ref is `refs/tags/` with no actual tag name.